File: src/uva/Leetcode00011.java
public class Leetcode00011 {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length -1;
        int max = 0;
        while (l < r) {
            max = Math.max(Math.min(height[l],height[r])*(r-l) ,max);
            if (height[l] > height[r]) {
                r--;
            } else {
                l++;
            }
        }
        return max;
    }
}

File: src/uva/Leetcode00011.swift
class Solution {
    func maxArea(height: [Int]) -> Int {
        var l = 0,  r = height.count-1, maxArea = 0
        while l <= r {
            var area = min(height[l], height[r])*(r-l) 
            maxArea = max(maxArea, area)
            if height[l] > height[r] {
                r -= 1
            } else if height[r] > height[l] {
                l += 1
            } else {
                l += 1
            }
        }
        return maxArea
    }
}
